addons/binding/org.openhab.binding.chamberlainmyq/src/main/java/org/openhab/binding/chamberlainmyq/handler/ChamberlainMyQGatewayHandler.java

```diff
@@ -0,0 +1,472 @@
+                "{\"ApplicationId\":\"%s\"," + "\"SecurityToken\":\"%s\"," + "\"MyQDeviceId\":\"%s\","
+                        + "\"AttributeName\":\"%s\"," + "\"AttributeValue\":\"%d\"}",
+                APP_ID, enc(tempToken), deviceID, name, state);
+        String url = String.format("%s/api/v4/DeviceAttribute/PutDeviceAttribute?appId=%s&SecurityToken=%s", WEBSITE,
+                enc(APP_ID), enc(tempToken));
+
+        request("PUT", url, message, "application/json", true, false, enc(tempToken));
+
+        if (rapidPoll) {
+            beginRapidPoll();
+        } else {
+            doFuturePoll(quickPoll * 1000);
+        }
+
+    }
+
+    public interface RequestCallback {
+        public void parseRequestResult(MyqJson resultJson);
+
+        public void onError(String error);
+    }
+
+    protected class Request implements Runnable {
+        private RequestCallback callback;
+
+        public Request(RequestCallback callback) {
+            this.callback = callback;
+        }
+
+        @Override
+        public void run() {
+            try {
+                MyqJson resultJson = getMyqData();
+                callback.parseRequestResult(resultJson);
+            } catch (Exception e) {
+                logger.warn("An exception occurred while executing a request to the MyQ Gateway: '{}'", e.getMessage());
+            }
+        }
+    }
+
+    public void sendRequestToServer(RequestCallback callback) throws IOException {
+        Request request = new Request(callback);
+        request.run();
+    }
+
+    private void startAutomaticRefresh() {
+        mainPollRefreshJob = scheduler.scheduleAtFixedRate(() -> {
+            try {
+                refreshDeviceState();
+            } catch (Exception e) {
+                logger.debug("Exception occurred during refresh: {}", e.getMessage(), e);
+            }
+        }, 0, refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private synchronized void refreshDeviceState() {
+        try {
+            MyqJson myqDevices = getMyqData();
+            Bridge bridge = getThing();
+
+            List<Thing> things = bridge.getThings();
+            for (Device myqdevice : myqDevices.getDevices()) {
+                String findDeviceId = myqdevice.getMyQDeviceId().toString();
+                for (Thing thing : things) {
+                    if (thing.getUID().getId().compareTo(findDeviceId) == 0) {
+                        ChamberlainMyQHandler test = (ChamberlainMyQHandler) thing.getHandler();
+                        if (test != null) {
+                            test.updateState(myqdevice);
+                        }
+                    }
+                }
+            }
+
+        } catch (Exception e) {
+            logger.warn("An exception occurred while executing a request to the Gateway: '{}'", e.getMessage());
+        }
+    }
+
+    // UTF-8 URL encode
+    private String enc(String str) {
+        try {
+            return URLEncoder.encode(str, "utf-8");
```

Comment:
Use `StandardCharsets.UTF_8.name()` [see](https://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html#UTF_8) this will allow you to remove the exception handling as well.